### PR TITLE
Add support for logging via Firelens

### PR DESCRIPTION
**Pull request by annuh**
This PR adds support to pass Firelens configurations via ecs-cli ecs-params.yml.
Needed to updated the aws-sdk which adds support for Firelens (https://github.com/aws/amazon-ecs-cli/pull/923)

More information about Firelens can be found here: 
https://github.com/aws/containers-roadmap/tree/master/preview-programs/firelens

Tested with the following files:

**docker-compose.yml**
```
version: '3'
services:
  log_router:
    image: amazon/aws-for-fluent-bit
    logging:
      driver: awslogs
      options:
        awslogs-group: test/logger
        awslogs-region: eu-central-1
        awslogs-stream-prefix: aws
  
  nginx:
    image: nginx
    logging:
      driver: awsfirelens
      options:
        Name: cloudwatch
        region: eu-central-1
        log_group_name: test/app
        auto_create_group: "true"
        log_stream_prefix: "app"
```

**ecs-params.yml**
```
version: 1
task_definition:
  task_role_arn: "ecsTaskRole"
  task_execution_role: ecsTaskExecutionRole
  ecs_network_mode: awsvpc
  task_size:
    mem_limit: 2GB
    cpu_limit: 1024
  services:
    log_router:
      firelens_configuration:
        type: fluentbit
        options:
          enable-ecs-log-metadata: "false"
run_params:
  network_configuration:
    awsvpc_configuration:
      subnets:
        - "subnet-xxx"
      security_groups:
        - "sg-xxx"
      assign_public_ip: ENABLED
```

Manually describing the task-definition gives:
```
➜ aws ecs describe-task-definition --task-definition log-from-fargate:5
{
    "taskDefinition": {
        "taskDefinitionArn": "arn:aws:ecs:eu-central-1:xxx:task-definition/log-from-fargate:5",
        "containerDefinitions": [
            {
                "name": "log_router",
                "image": "amazon/aws-for-fluent-bit",
                "cpu": 0,
                "links": [],
                "portMappings": [],
                "essential": true,
                "entryPoint": [],
                "command": [],
                "environment": [],
                "mountPoints": [],
                "volumesFrom": [],
                "linuxParameters": {
                    "capabilities": {},
                    "devices": []
                },
                "user": "0",
                "privileged": false,
                "readonlyRootFilesystem": false,
                "dnsServers": [],
                "dnsSearchDomains": [],
                "extraHosts": [],
                "dockerSecurityOptions": [],
                "pseudoTerminal": false,
                "logConfiguration": {
                    "logDriver": "awslogs",
                    "options": {
                        "awslogs-group": "test/logger",
                        "awslogs-region": "eu-central-1",
                        "awslogs-stream-prefix": "aws"
                    }
                },
                "firelensConfiguration": {
                    "type": "fluentbit",
                    "options": {
                        "enable-ecs-log-metadata": "false"
                    }
                }
            },
            {
                "name": "nginx",
                "image": "nginx",
                "cpu": 0,
                "links": [],
                "portMappings": [],
                "essential": true,
                "entryPoint": [],
                "command": [],
                "environment": [],
                "mountPoints": [],
                "volumesFrom": [],
                "linuxParameters": {
                    "capabilities": {},
                    "devices": []
                },
                "privileged": false,
                "readonlyRootFilesystem": false,
                "dnsServers": [],
                "dnsSearchDomains": [],
                "extraHosts": [],
                "dockerSecurityOptions": [],
                "pseudoTerminal": false,
                "logConfiguration": {
                    "logDriver": "awsfirelens",
                    "options": {
                        "Name": "cloudwatch",
                        "auto_create_group": "true",
                        "log_group_name": "test/app",
                        "log_stream_prefix": "app",
                        "region": "eu-central-1"
                    }
                }
            }
        ],
        "family": "log-from-fargate",
        "taskRoleArn": "arn:aws:iam::xxx:role/ecsTaskRole",
        "executionRoleArn": "arn:aws:iam::xxx:role/ecsTaskExecutionRole",
        "networkMode": "awsvpc",
        "revision": 5,
        "volumes": [],
        "status": "ACTIVE",
        "requiresAttributes": [
            {
                "name": "com.amazonaws.ecs.capability.logging-driver.awslogs"
            },
            {
                "name": "ecs.capability.execution-role-awslogs"
            },
            {
                "name": "com.amazonaws.ecs.capability.docker-remote-api.1.19"
            },
            {
                "name": "ecs.capability.firelens.fluentbit"
            },
            {
                "name": "com.amazonaws.ecs.capability.docker-remote-api.1.17"
            },
            {
                "name": "com.amazonaws.ecs.capability.logging-driver.awsfirelens"
            },
            {
                "name": "com.amazonaws.ecs.capability.task-iam-role"
            },
            {
                "name": "com.amazonaws.ecs.capability.docker-remote-api.1.18"
            },
            {
                "name": "ecs.capability.task-eni"
            }
        ],
        "placementConstraints": [],
        "compatibilities": [
            "EC2",
            "FARGATE"
        ],
        "requiresCompatibilities": [
            "FARGATE"
        ],
        "cpu": "1024",
        "memory": "2048"
    }
}
```

----
Enter `[N/A]` in the box, if an item is not applicable to your change.

**Testing**
- [x] Unit tests passed
- [x] Integration tests passed
- [x] Unit tests added for new functionality
- [x] Listed manual checks and their outputs in the comments ([example](https://github.com/aws/amazon-ecs-cli/pull/750#issuecomment-472623042))
- [ ] Link to issue or PR for the integration tests:

**Documentation**
- [ ] Contacted our doc writer
- [x] Updated our README
----

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by PettitWesley** [comment]
See https://github.com/aws/amazon-ecs-cli/pull/923#issuecomment-528964192

File: README.md [hunk 1]
```diff
@@ -544,6 +546,9 @@ Fields listed under `task_definition` correspond to fields that will be included
   * In Docker compose version 2, the `cpu_shares`, `mem_limit`, and `mem_reservation` fields can be specified in either the compose or ECS params file. If they are specified in the ECS params file, the values will override values present in the compose file.
   * If you are using a private repository for pulling images, `repository_credentials` allows you to specify an AWS Secrets Manager secret ARN for the name of the secret containing your private repository credentials as a `credential_parameter`.
   * `init_process_enabled` is a [Linux-specific option](https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_LinuxParameters.html) that can be be set to run an init process inside the container that forwards signals and reaps processes. This parameter maps to the `--init` option to [docker run](https://docs.docker.com/engine/reference/run/). This parameter requires version 1.25 of the Docker Remote API or greater on your container instance.
+  * `firelens_configuration` contains configuration parameters.
```
Comment (PettitWesley) [inline README.md]: nit: could link to the API docs here: https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_FirelensConfiguration.html

File: ecs-cli/modules/utils/compose/ecs_params_reader_test.go [hunk 2]
```diff
@@ -145,6 +151,9 @@ task_definition:
 
 		assert.ElementsMatch(t, expectedSecrets, wordpress.Secrets, "Expected secrets to match")
 		assert.ElementsMatch(t, expectedSecrets, wordpress.Logging.SecretOptions, "Expected secrets to match")
+
+		assert.Equal(t, "fluentbit", log_router.FirelensConfiguration.Type, "Exception firelens_configuration to be fluentbit")
+		assert.Equal(t, "true", log_router.FirelensConfiguration.Options["enable-ecs-log-metadata"], "Exception firelens_configuration to be fluentbit")
```
Comment (PettitWesley) [inline ecs-cli/modules/utils/compose/ecs_params_reader_test.go]: "Exception" => "Expected"

And the second one should say something like "Expected 'enable-ecs-log-metadata" to be 'true'"

File: ecs-cli/modules/utils/compose/convert_task_definition_test.go [hunk 3]
```diff
@@ -1755,58 +1755,48 @@ registry_credential_outputs:
...
+	assert.NoError(t, err, "Could not read ECS Params file")
+
+	containerConfigs := []adapter.ContainerConfig{*containerConfig}
+	taskDefinition, err := convertToTaskDefinitionForTest(t, containerConfigs, "", "", ecsParams, nil)
+
+	containerDefs := taskDefinition.ContainerDefinitions
+	log_router := findContainerByName("log_router", containerDefs)
+
+	if assert.NoError(t, err) {
+		assert.Equal(t, aws.String("fluentbit"), log_router.FirelensConfiguration.Type)
```
Comment (PettitWesley) [inline ecs-cli/modules/utils/compose/convert_task_definition_test.go]: Should add an assertion here for `enable-ecs-log-metadata` being `true`

**Comment by PettitWesley** [comment]
@annuh This looks good. 

Two final things:
1. Please post output of manually testing this change. Create a task definition and then describe it using the AWS CLI and post the output.
2. Please rebase this PR with dev, and rebase #923.

**Comment by annuh** [comment]
@PettitWesley I posted the resulting task definition into the PR's description. Also rebased my branches onto dev. Hope it's okay now 🙂 

This PR also included #923. Would you like to merge both PRs separately?

**Pull request merged**